lib/rules/no-array-constructor.js

```diff
@@ -81,6 +84,7 @@ module.exports = {
             if (
                 node.callee.type !== "Identifier" ||
                 node.callee.name !== "Array" ||
+                sourceCode.getTokenAfter(node.callee)?.value === "<" ||
```

Comment:
Is there a reason to dip into tokens rather than using the AST directly? This is what the existing typescript-eslint rule does:

```suggestion
                node.typeArguments ||
```

Reply:
was using `@typescript-eslint/parser` from astExplorer and it's nodes wasn't working so did this.